src/advanced_inv.cpp

```diff
@@ -622,8 +624,30 @@ void advanced_inventory::init()
     panes[right].sortby = ( advanced_inv_sortby ) uistate.adv_inv_rightsort;
     panes[left].area = ( aim_location ) uistate.adv_inv_leftarea;
     panes[right].area = ( aim_location ) uistate.adv_inv_rightarea;
-    bool moved = ( uistate.adv_inv_last_coords.x != g->u.posx() ||
-                   uistate.adv_inv_last_coords.y != g->u.posy() );
+
+    bool moved = ( uistate.adv_inv_last_coords.x != g->u.posx()
+                || uistate.adv_inv_last_coords.y != g->u.posy() 
+                || uistate.adv_inv_last_coords.z != g->u.posz());
+
+    // shorthand references
+    auto &pl = panes[left];
+    auto &pr = panes[right];
+    // finally! if it calls to load it, load it! (it being prior vehicle data)
+    if(pl.area == AIM_VEHICLE || pr.area == AIM_VEHICLE) {
+        // have we moved since last usage?
+        if(!moved) {
+            load_veh_data();
+        } else {
+            // reset pane that is AIM_VEHICLE
+            if(pl.area == AIM_VEHICLE)
```

Comment:
Braces around even single statements, not having braces is fragile because adding another statement may not do what you expect.